staging/src/k8s.io/client-go/examples/third-party-resources/main.go

```diff
@@ -78,19 +77,29 @@ func main() {
 	}
 
 	// make a new config for our extension's API group, using the first config as a baseline
-	var tprconfig *rest.Config
-	tprconfig = config
-	configureClient(tprconfig)
-
-	tprclient, err := rest.RESTClientFor(tprconfig)
+	exampleClient, exampleScheme, err := NewClient(config)
 	if err != nil {
 		panic(err)
 	}
 
+	// start a watcher on instances of our TPR
+	watcher := Watcher{
+		clientset:     clientset,
+		exampleClient: exampleClient,
+		exampleScheme: exampleScheme,
+	}
+
+	ctx, cancelFunc := context.WithCancel(context.Background())
+	defer cancelFunc()
+	go watcher.Run(ctx)
+
+	time.Sleep(5 * time.Second) // Wait until watcher starts
```

Comment:
Well, in the normal situation (not a hello world application) you wouldn't create an instance of TPR right after installing a watcher on it in the same app. So all this sleep is for is to wait until the watcher goroutine passes through the point of subscribing on updates. That's exactly what the comment is about.